packages/ag-grid-community/src/edit/editService.ts

```diff
@@ -36,13 +43,26 @@ import {
     _valuesDiffer,
     getCellEditorInstanceMap,
 } from './utils/editors';
+import { _getSiblingRows } from './utils/nodes';
 import { _refreshEditCells } from './utils/refresh';
 
 type BatchPrepDetails = { compDetails?: UserCompDetails; valueToDisplay?: any };
 
+// stop editing sources that we treat as UI-originated so we follow standard processing.
+const SOURCE_TRANSFORM: Record<string, EditSource> = {
+    paste: 'ui',
+    rangeSvc: 'ui',
+    fillHandle: 'api',
+    cellClear: 'api',
+};
+
+const SOURCE_TRANSFORM_KEYS: string[] = Object.keys(SOURCE_TRANSFORM);
+
+const CANCEL_PARAMS: StopEditParams = { cancel: true, source: 'api' };
+
 export class EditService extends BeanStub implements NamedBean, IEditService {
     beanName = 'editSvc' as const;
-    public batch: boolean;
+    private batch: boolean;
```

Comment:
```suggestion
    private batch: boolean = false;
```
Otherwise `isBatchEditing()` can return undefined

Reply:
Updated